File: OneLoginExample-OC/OneLoginExample/OneLogin/BaseViewController.m
//
//  BaseViewController.m
//  OneLoginExample
//
//  Created by noctis on 2019/11/21.
//  Copyright © 2019 geetest. All rights reserved.
//

#import "BaseViewController.h"

@interface BaseViewController ()

@end

@implementation BaseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (CGFloat)ol_screenWidth {
    return MIN([UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height);
}

- (CGFloat)ol_screenHeight {
    return MAX([UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height);
}

- (BOOL)isAuthViewControllerNotPresented:(NSDictionary *)result {
    if (result && result[@"errorCode"]) {
        NSString *errorCode = [NSString stringWithFormat:@"%@", result[@"errorCode"]];
        if ([errorCode isEqual:@"-20205"] ||
            [errorCode isEqual:@"-20206"] ||
            [errorCode isEqual:@"-20103"]) {
            return YES;
        }
        
        NSDictionary *metadata = result[@"metadata"];
        if (metadata && [metadata isKindOfClass:[NSDictionary class]] && metadata.count > 0) {
            NSString *resultCode = [NSString stringWithFormat:@"%@", metadata[@"resultCode"]];
            // 200022 无网络, 200023 请求超时, 200027 未开启数据网络
            if ([resultCode isEqual:@"200022"] ||
                [resultCode isEqual:@"200023"] ||
                [resultCode isEqual:@"200027"]) {
                return YES;
            }
        }
    }
    return NO;
}

- (BOOL)isIPhoneXScreen {
    if (@available(iOS 11.0, *)) {
        CGFloat safeAreaInsetBottom = [[UIApplication sharedApplication] delegate].window.safeAreaInsets.bottom;
        return safeAreaInsetBottom > 0.0;
    }
    
    return NO;
}

@end

File: OneLoginExample-Swift/OneLoginExample-Swift/OneLogin/BaseViewController.swift
//
//  SwiftBaseViewController.swift
//  OneLoginExample
//
//  Created by 刘练 on 2020/2/4.
//  Copyright © 2020 geetest. All rights reserved.
//

import UIKit

class BaseViewController: UIViewController {
    // OneLogin
    let GTOneLoginAppId = "b41a959b5cac4dd1277183e074630945"
    let GTOneLoginResultURL = "http://onepass.geetest.com/onelogin/result"
    
    // OnePass
    let GTOnePassAppId = "3996159873d7ccc36f25803b88dda97a"
    let GTOnePassVerifyURL = "http://onepass.geetest.com/v2.0/result"
        
    let NeedCustomAuthUI = true
    let OLAuthVCAutoLayout = true
        
    // MARK: ViewLifeCycle
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    // MARK: Screen
    
    func ol_screenHeight() -> CGFloat {
        return max(UIScreen.main.bounds.size.width, UIScreen.main.bounds.size.height)
    }
    
    func ol_screenWidth() -> CGFloat {
        return min(UIScreen.main.bounds.size.width, UIScreen.main.bounds.size.height)
    }
    
    func isIPhoneXScreen() -> Bool {
        if #available(iOS 11.0, *) {
            if let window = UIApplication.shared.delegate?.window, let bottom = window?.safeAreaInsets.bottom {
                return bottom > 0.0
            }
        }
        return false
    }
}
